backend/src/v2/driver/resolve.go

```diff
@@ -378,8 +379,22 @@ func resolveInputParameter(
 		default:
 			return nil, paramError(fmt.Errorf("param runtime value spec of type %T not implemented", t))
 		}
-	// TODO(Bobgy): implement the following cases
-	// case *pipelinespec.TaskInputsSpec_InputParameterSpec_TaskFinalStatus_:
+	case *pipelinespec.TaskInputsSpec_InputParameterSpec_TaskFinalStatus_:
+		tasks, err := getDAGTasks(ctx, dag, pipeline, mlmd, nil)
+		if err != nil {
+			return nil, err
+		}
+		producer, ok := tasks[opts.Task.DependentTasks[0]+"_"+strconv.FormatInt(dag.Execution.GetID(), 10)]
+		if !ok {
+			return nil, fmt.Errorf("producer task, %v, not in tasks", producer.TaskName())
+		}
+		finalStatusStruct, err := structpb.NewStruct(map[string]interface{}{
+			"state":                   producer.GetExecution().GetLastKnownState().String(),
+			"pipelineTaskName":        producer.TaskName(),
+			"pipelineJobResourceName": opts.RunName,
+			"error":                   map[string]interface{}{},
+		})
```

Comment:
Optional and untested: It's less efficient but I think this is safer since we are for sure using the correct fields and it's rendered in a struct that maps to correct JSON fields. I'll leave it up to you to decide which you prefer.

```suggestion
		finalStatus := pipelinespec.PipelineTaskFinalStatus{
			State: producer.GetExecution().GetLastKnownState().String(),
			PipelineTaskName: producer.TaskName(),
			PipelineJobResourceName: opts.RunName,
		}
		finalStatusJSON, err := json.Marshal(finalStatus)
		if err != nil {
			return nil, fmt.Errorf("failed to marshal PipelineTaskFinalStatus: %w", err)
		}

		var finalStatusMap map[string]interface{}
		if err := json.Unmarshal(finalStatusJSON, &finalStatusMap); err != nil {
			return nil, fmt.Errorf("failed to unmarshal JSON of PipelineTaskFinalStatus: %w", err)
		}

		// Convert the finalStatusMap to a structpb.Struct
		finalStatusStruct, err := structpb.NewStruct(finalStatusMap)
		if err != nil {
			return nil, fmt.Errorf("failed to create structpb.Struct: %w", err)
		}
		
		return structpb.NewStructValue(finalStatusStruct), nil
```